ballerina/llm_client_azure_open_ai.bal

```diff
@@ -24,41 +25,104 @@ public type AzureOpenAIModelConfig record {|
     string serviceUrl;
 |};
 
+type ChatCompletionAzureRequest record {
+    record {
+        "user"|"system" role;
+        string content;
+    }[] messages = [];
+    AzureOpenAIResponseFormat response_format;
+};
+
+type ChatCompletionRequest record {
+    string prompt;
+    AzureOpenAIResponseFormat responseFormat = {'type: "text"};
+};
+
+type AzureOpenAIResponseFormat ResponseFormatText|ResponseFormatJsonObject|ResponseFormatJsonSchema;
+
+type ResponseFormatText record {|
+    "text" 'type;
+|};
+
+type ResponseFormatJsonObject record {|
+    "json_object" 'type;
+|};
+
+type ResponseFormatJsonSchema record {|
+    "json_schema" 'type;
+    ResponseFormatJsonSchemaValue json_schema;
+|};
+
+type ResponseFormatJsonSchemaValue record {
+    string description?;
+    string name;
+    json schema?;
+    boolean? strict = true;
+};
+
+type ChatCompletionAzureResponse record {
+    AzureOpenAIResponseMessage[] choices?;
+};
+
+type AzureOpenAIResponseMessage record {
+    record {
+        string? content;
+    } message;
+};
```

Comment:
Why do we need all these types? Only define what is absolutely necessary. Make them closed if they are not required to be open (request types).